File: src/main/java/sunshine/common/ApiResponse.java
package sunshine.common;

import java.time.Instant;

/**
 * API 응답을 표준화하는 레코드.
 *
 * @param <T> 응답 데이터 타입
 */
public record ApiResponse<T>(
    boolean success,
    T data,
    ErrorResponse error,
    Instant timestamp
) {
  public static <T> ApiResponse<T> ok(T data) {
    return new ApiResponse<>(true, data, null, Instant.now());
  }

  public static <T> ApiResponse<T> error(ErrorResponse error) {
    return new ApiResponse<>(false, null, error, Instant.now());
  }
}

File: src/main/kotlin/sunshine/common/ApiResponse.kt
package sunshine.common

import java.time.Instant

/**
 * API 응답을 표준화하는 레코드.
 */
data class ApiResponse<T>(
    val success: Boolean,
    val data: T?,
    val error: ErrorResponse?,
    val timestamp: Instant
) {
    companion object {
        fun <T> ok(data: T): ApiResponse<T> = ApiResponse(true, data, null, Instant.now())
        fun <T> error(error: ErrorResponse): ApiResponse<T> = ApiResponse(false, null, error, Instant.now())
    }
}
